lib/request-proxy.js

```diff
@@ -86,158 +76,77 @@ proto.proxyReq = function proxyReq(opts) {
     var dest = opts.dest;
     var req = opts.req;
     var res = opts.res;
+
+    // Request-level overrides (see ctor).
     var maxRetries = opts.maxRetries;
     var retrySchedule = opts.retrySchedule;
     var endpoint = opts.endpoint || '/proxy/req';
-
-    var ringpop = this.ringpop;
-
     var timeout = opts.timeout ?
-        opts.timeout : ringpop.proxyReqTimeout;
+        opts.timeout : this.ringpop.proxyReqTimeout;
```

Comment:
nitpick: consider using `self` and `this` consistently.

You use `this` in the function and `self` in the `onBody` closure.